File: string/increasing-decreasing-string/increasing-decreasing-string.go
package increasingdecreasingstring

func sortString(s string) string {
	count := [26]rune{}
	for _, c := range s {
		count[c-'a']++
	}

	ans := make([]rune, 0, len(s))

	for len(ans) != len(s) {
		for i := 0; i < 26; i++ {
			if count[i] > 0 {
				count[i]--
				ans = append(ans, rune(i+'a'))
			}
		}
		for i := 25; i >= 0; i-- {
			if count[i] > 0 {
				count[i]--
				ans = append(ans, rune(i+'a'))
			}
		}
	}

	return string(ans)
}

File: string/increasing-decreasing-string/increasing-decreasing-string.ts
function sortString(s: string): string {
    const count = new Array<number>(26).fill(0);
    for (let c of s) {
        count[c.charCodeAt(0) - 'a'.charCodeAt(0)]++;
    }

    let ans: string[] = [];

    while (ans.length !== s.length) {
        for (let i = 0; i < 26; i++) {
            if (count[i] > 0) {
                ans.push(String.fromCharCode(i + 'a'.charCodeAt(0)))
                count[i]--;
            }
        }
        for (let i = 25; i >= 0; i--) {
            if (count[i] > 0) {
                ans.push(String.fromCharCode(i + 'a'.charCodeAt(0)))
                count[i]--;
            }
        }
    }

    return ans.join('');
};
